ipa-core/src/protocol/context/batcher.rs

```diff
@@ -111,13 +116,14 @@ impl<'a, B> Batcher<'a, B> {
     fn get_batch_by_offset(&mut self, batch_offset: usize) -> &mut BatchState<B> {
         if self.batches.len() <= batch_offset {
             self.batches.reserve(batch_offset - self.batches.len() + 1);
+            let pending_records_capacity = self.records_per_batch.min(TARGET_PROOF_SIZE);
```

Comment:
Note that `TARGET_PROOF_SIZE` is not really valid here -- what we really want is `TARGET_PROOF_SIZE / steps_in_this_protocol`. But it doesn't seem worth going down that road, because what we really really want is for `records_per_batch` to have a useful value.